Created an image manipulation cog.

bsrecker opened the pull request:
Commands:
```
-Filter - Filters images using the following
-blend - Blends two images together based on some alpha 
-colors - retrieve all the hexcodes in the image and their counts. 
-grayscale - Convert image to grayscale.
-solarize - Solarize grayscale images above some threshold.
```

Note: This is just a start more commands can be added based on feedback.

bsrecker committed:
Created an image manipulation cog.

Commands:
-Filter - Filters images using the following
-blend - Blends two images together based on some alpha
-colors - retrieve all the hexcodes in the image and their counts.
-grayscale - Convert image to grayscale.
-solarize - Solarize grayscale images above some threshold.

bsrecker committed:
Add extension to config.

bsrecker committed:
Add extension to config.

mudkipdev commented:
Why should it be for the v0.2.0 milestone? I'm not against adding an image manipulation cog, but I was hoping to do these things post-release at the very least.

mudkipdev commented on leaf/config.toml:
```diff
@@ -3,8 +3,9 @@ token = ""
 [database]
-connection_uri = ""
\ No newline at end of file
+connection_uri = "postgresql://postgres:ginger18!@localhost"
```
You probably did not mean to include this.

mudkipdev commented on leaf/extensions/image.py:
```diff
@@ -0,0 +1,220 @@
+import enum
+
+import PIL.Image
```
Is PIL not blocking? Also what is the point of importing `PIL.Image` when you've already used `from PIL import Image`?

bsrecker replied:
Fixed. And yes, it might be blocking but I think these calls are generally pretty fast so shouldn't actually cause too much of an issue. More expensive commands might cause more of a problem

bsrecker replied:
We could run some of these commands in an executor if they become problematic.

Infernum1 replied:
Yeah as far as I'm aware there's no async version for pillow

mudkipdev commented on leaf/extensions/image.py:
```diff
@@ -0,0 +1,220 @@
+from discord.ext import commands
+from discord import app_commands
+import discord
+from bot import LeafBot
```
Remember to follow PEP 8 order, standard library imports first, then local imports, then third-party libraries.

mudkipdev reviewed (changes requested):
My surface-level thoughts so far. As for the filters, there could probably be more and I will give my ideas later.

bsrecker committed:
Update config.toml

bsrecker committed:
Resolve initial review issues.

bsrecker committed:
Merge remote-tracking branch 'origin/Image-Manipulation' into Image-Manipulation

bsrecker commented:
> Why should it be for the v0.2.0 milestone? I'm not against adding an image manipulation cog, but I was hoping to do these things post-release at the very least.

That's up to you. This was just something I was interested in. It doesn't need to go with v1.

bsrecker committed:
Button permissions.

bsrecker committed:
Refactor code to make it more simple.
Disable buttons after timeout.
Check for disabled button interactions and return early.

bsrecker committed:
Cleaner button labels.

bsrecker committed:
Move config back

bsrecker committed:
Move config back

bsrecker committed:
Delete config.toml

bsrecker committed:
Move config back

bsrecker committed:
Merge remote-tracking branch 'origin/Image-Manipulation' into Image-Manipulation

# Conflicts:
#	config.toml

bsrecker committed:
Move config back

bsrecker committed:
Merge branch 'mudkipdev:main' into Image-Manipulation

bsrecker committed:
Bug Fix

bsrecker committed:
Merge remote-tracking branch 'origin/Image-Manipulation' into Image-Manipulation

bsrecker committed:
Bug Fix

bsrecker committed:
Apply GPG key

bsrecker committed:
Apply GPG key

bsrecker committed:
Delete image.py

mudkipdev commented:
Why was this deleted?
